### PR TITLE
[JAX] Fix mesh resource requirement when no mesh

### PR DESCRIPTION
**Pull request by jberchtold-nvidia**
# Description

When not using Flax logical axes, a global MeshResource context was required by TE even when no JAX mesh was active. This PR fixes this case and avoids raising an unnecessary error.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

- Update TE's logical axis system to return an empty mapping and avoid raising an error when no JAX mesh is active. Use case, new user to TE trying it out on single-GPU. Want to avoid unnecessary complexity in cases where it isn't required

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/jax/sharding.py [hunk 1]
```diff
@@ -80,6 +80,11 @@ def get_sharding_map_logic_axis_to_mesh_axis():
         # If no mesh is defined, return an empty dict and do not require a MeshResource context to be present
         return {}
 
+    abstract_mesh = get_abstract_mesh()
+    if abstract_mesh is None or sorted(abstract_mesh.manual_axes) == sorted(mesh.axis_names):
```
Comment (phu0ngng) [inline transformer_engine/jax/sharding.py]: Hi, why do we need to check for `if abstract_mesh is None` here?

Comment (jberchtold-nvidia) [reply transformer_engine/jax/sharding.py]: I wasn't exactly sure how the abstract mesh relates to the PXLA physical mesh above. If they both are always set via `with jax.sharding.Mesh():`, then this `abstract_mesh is None` check is redundant.

Comment (phu0ngng) [reply transformer_engine/jax/sharding.py]: Looks like the `get_abstract_mesh()` won't return None.
https://github.com/jax-ml/jax/blob/eace2086ab710c2b95b40e926a44212d73cfd7ae/jax/_src/mesh.py#L625

Comment (jberchtold-nvidia) [reply transformer_engine/jax/sharding.py]: Thanks @phu0ngng! I've removed the `abstract_mesh is None` check in the latest version
